File: lib/data/models/recipe/recipe_search_model.dart
import 'package:calory_tool/data/models/recipe/recipe_model.dart';

class RecipeSearchModel {
  const RecipeSearchModel({this.pageNumber, this.recipes});

  factory RecipeSearchModel.fromJson(Map<String, dynamic> json) {
    return RecipeSearchModel(
      pageNumber: json['page_number'] as int?,
      recipes:
          (json['recipes'] as List<dynamic>?)
              ?.map((e) => RecipeModel.fromJson(e as Map<String, dynamic>))
              .toList(),
    );
  }

  final int? pageNumber;
  final List<RecipeModel>? recipes;
}

File: server/src/models/recipes/RecipeSearchModel.js
import RecipeModel from "./RecipeModel.js";

class RecipeSearchModel {
  constructor({ pageNumber = null, recipeModel = null } = {}) {
    this.pageNumber = pageNumber;
    this.recipeModel = recipeModel;
  }

  static fromJson(json) {
    const recipesJson = json["recipes"];
    const recipeJson = recipesJson ? recipesJson["recipe"] : null;
    const recipes = recipeJson ? recipeJson.map(e => RecipeModel.fromJson(e)) : [];
    return new RecipeSearchModel({
      pageNumber: recipesJson?.["page_number"] ? parseInt(recipesJson["page_number"]) : null,
      recipes: recipes,
    });
  }
}

export default RecipeSearchModel;
